### ISSUE TITLE
Raising StopIteration from inside a task causes a RuntimeError

### ISSUE THREAD

**Issue by johnxx** [WarriorOfWire/CircuitPython_async#16]
This is my first time dabbling with async/await-style programming in Python, so if I'm simply failing to understand something here, please feel free to tell me to RTFM. :)

With that caveat: From reading the source code for `asynccp/loop.py`, I was under the impression I could `raise StopIteration` from inside the `coroutine_function` passed to a `ScheduledTask` to achieve the equivalent of calling `my_task.stop()`. However, when I actually `raise StopIteration` I get a stack trace:
```
Traceback (most recent call last):
  File "code.py", line 307, in <module>
  File "/lib/asynccp/loop.py", line 268, in run
  File "/lib/asynccp/loop.py", line 275, in _step
  File "/lib/asynccp/loop.py", line 303, in _run_task
  File "/lib/asynccp/loop.py", line 101, in _run_at_fixed_rate
RuntimeError: generator raised StopIteration

```
I believe that somewhere between raising StopIteration and the exception being handled in _run_task it's being turned into a `RuntimeError`.

I'm currently running this on `Adafruit CircuitPython 7.0.0 on 2021-09-20; TinyS2 with ESP32S2`

Let me know if some sample code (or any other information) would be helpful.

**Comment by kvc0** [comment]
![30265BBA-0A89-4D82-A659-103C1B3FB6FB](https://user-images.githubusercontent.com/3454741/146582528-96ac34d6-6095-4016-bfe6-666592aacc29.png)

Here I tried the same thing on my phone with CPython. The behavior you describe is exactly what is expected; I didn’t make the exception message match cpython in the coroutine case but I did make the type match (RuntimeError). That’s down in the guts of the async/await circuitpython code. If you’re feeling plucky you might go take a dive and see =)
